whois-update/src/main/java/net/ripe/db/whois/update/mail/MailGatewaySmtp.java

```diff
@@ -5,84 +5,90 @@
-import net.ripe.db.whois.update.domain.ResponseMessage;
-import net.ripe.db.whois.update.log.LoggerContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
-import org.springframework.mail.MailException;
-import org.springframework.mail.MailSendException;
 import org.springframework.mail.javamail.JavaMailSender;
 import org.springframework.mail.javamail.MimeMessageHelper;
-import org.springframework.stereotype.Component;
 
 import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-@Component
-public class MailGatewaySmtp implements MailGateway {
+public abstract class MailGatewaySmtp {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(MailGatewaySmtp.class);
+    protected static final Pattern INVALID_EMAIL_PATTERN = Pattern.compile("(?i)((?:auto|test)\\-dbm@ripe\\.net)");
 
-    private static final Pattern INVALID_EMAIL_PATTERN = Pattern.compile("(?i)((?:auto|test)\\-dbm@ripe\\.net)");
+    protected final MailConfiguration mailConfiguration;
+    protected final JavaMailSender mailSender;
+    protected final EmailStatusDao emailStatusDao;
+    protected final OutgoingMessageDao outgoingMessageDao;
+    protected final String webBaseUrl;
 
-    private final LoggerContext loggerContext;
-    private final MailConfiguration mailConfiguration;
-    private final JavaMailSender mailSender;
-    private final EmailStatusDao emailStatusDao;
-    private final OutgoingMessageDao outgoingMessageDao;
-    private final String webBaseUrl;
 
-    @Autowired
     public MailGatewaySmtp(
-            final LoggerContext loggerContext,
             final MailConfiguration mailConfiguration,
             final JavaMailSender mailSender,
             final EmailStatusDao emailStatusDao,
             final OutgoingMessageDao outgoingMessageDao,
             @Value("${web.baseurl}") final String webBaseUrl) {
-        this.loggerContext = loggerContext;
         this.mailConfiguration = mailConfiguration;
         this.mailSender = mailSender;
         this.emailStatusDao = emailStatusDao;
         this.outgoingMessageDao = outgoingMessageDao;
         this.webBaseUrl = webBaseUrl;
     }
 
-    @Override
-    public void sendEmail(final String to, final ResponseMessage responseMessage) {
-        sendEmail(to, responseMessage.getSubject(), responseMessage.getMessage(), responseMessage.getReplyTo());
+    protected abstract boolean canNotSendEmail(final String emailAddresses);
+
+    @Nullable
+    protected String extractEmailBetweenAngleBrackets(final String email) {
+        if(email == null) {
+            return null;
+        }
+
+        try {
+            return new InternetAddress(email).getAddress();
+        } catch (AddressException e) {
+            return email;
```

Comment:
In what cases it can throw AddressException ? I think if should cehck for angle brackets only thn try to parse???

Reply:
When is not able to parse the address seems.

    public InternetAddress(String address) throws AddressException {
        InternetAddress[] a = parse(address, true);
        if (a.length != 1) {
            throw new AddressException("Illegal address", address);
        } else {
            this.address = a[0].address;
            this.personal = a[0].personal;
            this.encodedPersonal = a[0].encodedPersonal;
        }
    }

This is already in master. This change was suggested by Ed. Using that library instead of just removing the angle brackets